service/collector.go

```diff
@@ -183,7 +183,15 @@ LOOP:
 			col.logger.Info("Received shutdown request")
 			break LOOP
 		case <-ctx.Done():
-			col.logger.Error("Context done, terminating process", zap.Error(ctx.Err()))
+			col.logger.Info("Context done, terminating process")
+
+			// Log error for when a context has timed out.
+			// A context that is canceled could be do to intercepted signals or intentional
+			// and we don't want to log an error with a stack trace to the user.
+			err := ctx.Err()
+			if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Why is this error special?

Reply:
I figured you still wanted to know if the collector was stopped due to a DeadlineExceed error as that would be less expected than a cancel. I think though the above comment of maybe just logging the error with no stack trace would allow us to remove this.